org.eclipse.xtext.builder/src/org/eclipse/xtext/builder/BuilderParticipant.java

```diff
@@ -590,9 +587,13 @@ protected void handleChangedContents(Delta delta, IBuildContext context, Eclipse
 		saveResourceStorage(resource, fileSystemAccess);
 		if (shouldGenerate(resource, context)) {
 			try {
+				IProgressMonitor fsaMonitor = fileSystemAccess.getMonitor();
 				MonitorBasedCancelIndicator cancelIndicator = new MonitorBasedCancelIndicator(fileSystemAccess.getMonitor());
 				GeneratorContext generatorContext = new GeneratorContext();
 				generatorContext.setCancelIndicator(cancelIndicator);
+				// Since we make use of SubMonitor.split instead of SubMonitor.newChild
+				// there is no reason anymore to avoid showing each resource.
+				fsaMonitor.subTask("Compile " + resource.getURI());
```

Comment:
The URI is not necessarily nice to read. We should probably use only the last segment.

Reply:
Done.